blocks/carousel/carousel.js

```diff
@@ -0,0 +1,211 @@
+  // decorate CTA
+  col.querySelectorAll('a').forEach((cta) => cta.classList.add('button'));
+
+  // decorate top & bottom content
+  col.childNodes.forEach((node) => {
+    if (node.nodeType === Node.ELEMENT_NODE && node.tagName.toLowerCase() === 'hr') {
+      // if node is an <hr> switch to the bottom section
+      isTop = false;
+    } else if (isTop) {
+      top += node.outerHTML || node.textContent;
+    } else {
+      bottom += node.outerHTML || node.textContent;
+    }
+  });
+
+  if (top.trim()) {
+    const $top = div({ class: 'content top' });
+    $top.innerHTML = top.trim();
+    frag.append($top);
+  }
+
+  if (bottom.trim() || !isTop) {
+    const $bottom = div({ class: 'content bottom' });
+    $bottom.innerHTML = bottom.trim();
+    frag.append($bottom);
+  }
+
+  return frag;
+}
+
+function createSlide(row, i) {
+  const isFirst = i === 1;
+  const $slide = li({ 'data-slide-index': i, class: `slide ${isFirst ? 'active' : ''}` });
+
+  row.querySelectorAll(':scope > div').forEach((col, c) => {
+    // decorate image
+    if (c === 0) {
+      col.classList.add('slide-image');
+      const img = col.querySelector('img');
+      if (img) {
+        const imgSizes = [
+          { media: '(max-width: 480px)', width: '480' },
+          { media: '(min-width: 480px) and (max-width: 768px)', width: '768' },
+          { media: '(min-width: 768px) and (max-width: 1024px)', width: '1024' },
+          { media: '(min-width: 1024px)', width: '1920' },
+        ];
+        col.innerHTML = '';
+        col.append(createOptimizedPicture(img.src, img.alt || `slide ${c}`, true, imgSizes));
+        $slide.append(col);
+      }
+    }
+    // decorate content
+    if (c === 1) {
+      // use default content if col is empty
+      const content = (col.textContent === '') ? defaultContent.cloneNode(true) : decorateSlideContent(col);
+      $slide.append(content);
+    }
+  });
+  return $slide;
+}
+
+export default async function decorate(block) {
+  const autoClass = block.className.split(' ').find((className) => className.startsWith('auto-'));
+
+  // get duration from auto class
+  if (autoClass) {
+    [autoDuration] = autoClass.match(/\d+/);
+    isAuto = true;
+  }
+
+  block.setAttribute('role', 'region');
+  block.setAttribute('aria-roledescription', 'Carousel');
+
+  const rows = block.querySelectorAll(':scope > div');
+  const isMultiple = rows.length > 2;
+  const $slides = ul({ class: 'slides' });
+
+  rows.forEach((row, i) => {
+    // set default content
+    if (i === 0 && isMultiple) {
+      defaultContent = decorateSlideContent(row.querySelector(':scope > div').nextElementSibling);
+    } else {
+      $slides.appendChild(createSlide(row, !isMultiple ? 1 : i));
+    }
+  });
+
+  const $container = div({ class: 'slides-container' },
+    $slides,
+  );
+
+  // add buttons if multiple slides
+  if (isMultiple) {
+    block.dataset.activeSlide = 0;
+    const $prev = button({ class: 'prev', 'aria-label': 'Previous Slide' });
```

Comment:
There's something about the buttons such that when you give the area focus, the :focus works and adjusts the transparency correctly, however clicking on the arrow breaks the transparency such that subsequent clicks do not activate the :focus or :hover styles adjusting the transparency correctly.